ggml/src/ggml-sycl/quants.hpp

```diff
@@ -0,0 +1,61 @@
+//
+// MIT license
+// Copyright (C) 2025 Codeplay Software Ltd.
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: MIT
+//
+
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+
+#ifndef GGML_SYCL_QUANTS_HPP
+#define GGML_SYCL_QUANTS_HPP
+
+#include "ggml-common.h"
+#include "ggml.h"
+
+namespace ggml_sycl_reordered {
+
+
+// The reordered block moves quants (qs) and  scales(d) to two
+// uniform regions of memory that is contiguous in the same tensor.
+// What this means is that instead of having:
+// [d0, qs0] [d1, qs1] [d2, qs2] ... [dN, qsN]
+// We have:
+// [qs0, qs1, qs2, ..., qsN]  [d0, d1, d2, ..., dN]
+//
+// Notes: out-of-bounds qs will run into d values
+// Aligment relies on the allocated size of qs
+
+template <ggml_type type> struct block_q_t;
+
+
+// qk number of weights / quants in a block
+// qr number of weights in a byte (described as 'before dequantization')
+//    for quantization types that has low and high bits split, qr is calculated with
+//    using the lower bits, e.g for Q6 quants QR6 is 2
+// qi size of a block in 32 bit integers
```

Comment:
```suggestion
// qi number of 32 bit integers needed to represent all the quants from a block (`qs` field)
```

This is more inline with my understanding of `qi`, thoughts?

Reply:
I think it's the number of quants that can be packed in a 32 bit type, that's why from [ggml-common.h](https://github.com/ggml-org/llama.cpp/blob/fccf9cae83e6c6cd31a0ecb403d237638e427d0a/ggml/src/ggml-common.h#L96),the value of QI4_0 (which is `(QK4_0 / (4 * QR4_0))`) comes out to be 8, as each quant in this case is 4 bit, thus 2 per byte and therefore 4 * 2 = 8 in a 32 bit type..

Number of 32 bit integer sounds like more than one integer could be required, which I do not think that' what this is field is trying to convey

Reply:
A lot clearer, I agree.